### PR TITLE
Adds documentation for Mesh objects

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #8329)

### CODE REVIEW

File: docs/content/documentation/systems/Mesh/index.md [hunk 1]
```diff
@@ -1,7 +1,83 @@
...
+
+Human-readable names can be assigned to blocks, sidesets, and nodesets. These names will be automatically read in and
+can be used throughout the input file. Mesh generators such as CUBIT will generally provide the capability internally.
+Any parameter that takes entity IDs in the input file will accept either numbers or "names". Names can also be
+assigned to IDs on-the-fly in existing meshes to ease input file maintenance (see example). On-the-fly names will
+also be written to Exodus/XDA/XDR files. An illustration for mesh in exodus file format.
+
+
+```text
+[Mesh]
```
Comment (dschwen) [inline docs/content/documentation/systems/Mesh/index.md]: Soo... whatabout taking that block from a tested file in the repo...

File: docs/content/documentation/systems/Mesh/index.md [hunk 2]
```diff
@@ -1,7 +1,83 @@
...
+all simulations. In parallel, the replicated format copies the complete mesh to all processors allowing for efficient
+access to the geometry elements.
+
+The distributed mesh format is useful when the mesh data structure dominates memory usage. Only the pieces of the
+mesh "owned" by a processor are actually stored on the processor. If the mesh is too large to read in on a single
+processor, it can be split prior to the simulation.
+
+!!! note
+    Both the "replicated" and "distributed" mesh formats are parallel with respect to the execution of the finite
+    element assembly and solve.
```
Comment (dschwen) [inline docs/content/documentation/systems/Mesh/index.md]: maybe point out that the solution data is always distributed and makes up a larger chunk of the memory usage in most simulations.

**Review by dschwen — APPROVED** [review]

**Comment by dschwen** [comment]
@brianmoose this is the second time I'm seeing the `/moose/.git/index.lock': File exists` failure. Any idear why that's happening?

**Comment by dschwen** [comment]
What's going on on the build boxes?!

**Comment by permcody** [comment]
Disk is full

**Comment by brianmoose** [comment]
Not sure what the `.git/index.lock` is all about. It seems to be caused by git crashing in the middle of an operation so doesn't remove the lock. No idea why that would happen though.
I have seen this several times. It seems to just have started about a week ago...

**Comment by dschwen** [comment]
Git might die because the disk space runs out.

**Review by dschwen — APPROVED** [review]

**Pull request merged**